Mage.Sets/src/mage/cards/r/RatadrabikOfUrborg.java

```diff
@@ -0,0 +1,110 @@
+
+    private static final FilterControlledCreaturePermanent filter2 = new FilterControlledCreaturePermanent("another legendary creature you control");
+
+    static {
+        filter.add(SubType.ZOMBIE.getPredicate());
+        filter2.add(SuperType.LEGENDARY.getPredicate());
+        filter2.add(AnotherPredicate.instance);
+    }
+
+    public RatadrabikOfUrborg(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{W}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ZOMBIE);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Ward {2}
+        this.addAbility(new WardAbility(new ManaCostsImpl<>("{2}")));
+
+        // Other zombies you control have vigilance.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityAllEffect(
+                VigilanceAbility.getInstance(), Duration.WhileOnBattlefield, filter, true
+        )));
+
+        // Whenever another legendary creature you control dies, create a token that's a copy of that creature,
+        // except it's not legendary and it's a 2/2 black Zombie in addition to its other colors and types.
+        this.addAbility(new DiesCreatureTriggeredAbility(new RatadrabikOfUrborgEffect(),false, filter2, true));
+
+    }
+
+    private RatadrabikOfUrborg(final RatadrabikOfUrborg card) {
+        super(card);
+    }
+
+    @Override
+    public RatadrabikOfUrborg copy() {
+        return new RatadrabikOfUrborg(this);
+    }
+}
+
+class RatadrabikOfUrborgEffect extends OneShotEffect {
+
+    public RatadrabikOfUrborgEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "create a token that's a copy of that creature, " +
+                "except it's not legendary and it's a 2/2 black Zombie in addition to its other colors and types.";
+    }
+
+    public RatadrabikOfUrborgEffect(final RatadrabikOfUrborgEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RatadrabikOfUrborgEffect copy() {
+        return new RatadrabikOfUrborgEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent copyFrom = targetPointer.getFirstTargetPermanentOrLKI(game, source);
+        if(controller == null || copyFrom == null) {
+            return false;
+        }
+        CreateTokenCopyTargetEffect effect = new CreateTokenCopyTargetEffect(controller.getId(), null, false,1,false,false,null,2,2,false);
+        effect.setAdditionalSubType(SubType.ZOMBIE);
+        effect.setIsntLegendary(true);
+        effect.setTargetPointer(new FixedTarget(copyFrom.getId(),game));
+        ObjectColor colors = copyFrom.getColor(game);
```

Comment:
I think this should be `.getColor()` without the game parameter. You want the original color(s) of the card only. Passing `game` will also copy any colors gained due to another effect (e.g. [[Blade of the Oni]]